File: sources/ruby/06_char_count/main.rb
# task 06 char_count — expected output: 10000000
# build: none (interpreted)    run: ruby main.rb (cruby) | ruby --yjit main.rb (cruby+yjit) | jruby main.rb (jruby, needs Java 25)
# The stock Windows CRuby build has no YJIT: `ruby --yjit` warns "Ruby was built without YJIT support".

text = 'abcdefghij' * 10_000_000

count = 0
text.each_char do |ch|
  if ch == 'a'
    next
  elsif ch == 'e'
    next
  elsif ch == 'h'
    count += 1
  end
end

puts count

File: sources/php/06_char_count/main.php
<?php
// task 06 char_count — expected output: 10000000
// build: none (interpreted)    run: php main.php (zend) | php -d opcache.enable_cli=1 -d opcache.jit_buffer_size=64M main.php (zend + jit)
// The 100 MB text is built once with str_repeat; the scan is an index loop over its bytes.

$text = str_repeat('abcdefghij', 10000000);

$count = 0;
$len = strlen($text);

for ($i = 0; $i < $len; $i++) {
    $ch = $text[$i];
    if ($ch === 'a') {
        continue;
    } elseif ($ch === 'e') {
        continue;
    } elseif ($ch === 'h') {
        $count++;
    }
}

echo $count, "\n";
